cider-velox/src/planTransformer/PatternRewriter.h

```diff
@@ -24,17 +24,31 @@
 #include "PlanBranches.h"
 #include "PlanPattern.h"
 #include "PlanRewriter.h"
+#include "../ciderTransformer/CiderPlanPatterns.h"
 
 namespace facebook::velox::plugin::plantransformer {
+
 class PatternRewriter {
  public:
   PatternRewriter(std::shared_ptr<PlanPattern> pattern,
                   std::shared_ptr<PlanRewriter> rewriter)
       : pattern_(pattern), rewriter_(rewriter) {}
 
+  std::shared_ptr<PlanPattern> clonePattern(std::shared_ptr<PlanPattern> ptr) const {
```

Comment:
More comfirtable way here is to implement a clone for every PlanPattern and interator the list here to clone them. Don't hardcode the registered pattern here since there should be a single place to specify the reigistered pattern. And also as we talked ago, here,what we actually need is the type info of the pattern and based on the type info to produce a new instance not cloning the existing instance. Pls cnsider change it later.

Reply:
Yes, I have considered using c++ type template type traits to implement a function like this:
```
using PatternType = extract_pattern_type<BasePlanPattern>;
std::shared_type<PatternType> p = std::make_shared<PatternType>();
```
however, to implement function extract_pattern_type, more codes than this pr will be changed or refactored.
so I just modify here with a simple way to make all TPC-H running correctly.